### PR TITLE
feat(jans-auth-server): OpenID AuthZEN implementation

### LINKED ISSUE #9557: feat(jans-auth-server): OpenID AuthZEN
### Description

OpenID AuthZEN is in the first implementers draft:  https://openid.github.io/authzen/

This issue is a placeholder to track all the various tasks that would be involved to implement support for this new spec. 

The backend for this service would be a Cedarling with a certain policy store loaded.

### PR DESCRIPTION
**Pull request by yuriyz**
### Description

feat(jans-auth-server): OpenID AuthZEN implementation

#### Target issue
  
closes #9557

### Test and Document the changes
- [ ] Static code analysis has been run locally and issues have been fixed
- [ ] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [ ] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

File: jans-core/script/src/main/java/io/jans/model/custom/script/type/authzen/DummyAccessEvaluationType.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        return true;
+    }
+
+    @Override
+    public int getApiVersion() {
+        return 1;
+    }
+
+    @Override
+    public AccessEvaluationResponse evaluate(AccessEvaluationRequest request, Object context) {
```
Comment (nynymike) [inline jans-core/script/src/main/java/io/jans/model/custom/script/type/authzen/DummyAccessEvaluationType.java:34]: What if you want to return "diagnostics" or some other type of error detail about why the policy failed.

File: jans-linux-setup/jans_setup/templates/scopes.ldif [hunk 2]
```diff
@@ -119,6 +119,17 @@ jansScopeTyp: openid
 objectClass: top
 objectClass: jansScope
 
+dn: inum=6D93,ou=scopes,o=jans
+description: Access Evaluation
+displayName: access_evaluation
+inum: 6D93
+jansAttrs: {"spontaneousClientId":"","spontaneousClientScopes":[],"showInConfigurationEndpoint":true}
```
Comment (nynymike) [inline jans-linux-setup/jans_setup/templates/scopes.ldif:126]: Do you reall need `spontaneousClientId`, `spontaneousClientScopes`, and `showInConfigurationEndpoint` here?

File: jans-core/script/src/main/java/io/jans/model/custom/script/type/authzen/DummyAccessEvaluationType.java (hunk 1 above)
Comment (yuriyz) [reply jans-core/script/src/main/java/io/jans/model/custom/script/type/authzen/DummyAccessEvaluationType.java:34]: @nynymike Script has to throw exception with all details inside. Example:
```
    @Override
    public AccessEvaluationResponse evaluate(AccessEvaluationRequest request, Object scriptContext) {

        if (request.getResource().getType().equalsIgnoreCase("file")) {
            throw new WebApplicationException(Response
                    .status(Response.Status.BAD_REQUEST)
                    .entity("{\n" +
                            "  \"error\": \"invalid_resource_type\",\n" +
                            "  \"error_description\": \"Resource type 'file' is not allowed.\"\n" +
                            "}")
                    .build());
        }
        ...
}        
```

File: jans-linux-setup/jans_setup/templates/scopes.ldif (hunk 2 above)
Comment (yuriyz) [reply jans-linux-setup/jans_setup/templates/scopes.ldif:126]: @nynymike this is default template values. AS reads it but it's not needed for AuthZEN if that was the question.

**Review by yuriyzz — APPROVED** [review]

**Review by yurem — APPROVED** [review]

**Comment by yuriyz** [comment]
Closed in favor of https://github.com/JanssenProject/jans/pull/10197
We had to get rid of unsigned commits after git history clean up.

**Pull request closed without merging**